python/tests/tests.py

```diff
@@ -42,7 +42,7 @@ class PythonUnitTestCase(unittest.TestCase):
 class TestSparkContext(object):
     @classmethod
     def setup_env(cls):
-        cls.sc = SparkContext('local[*]', cls.__name__)
+        cls.sc = SparkContext.getOrCreate()
```

Comment:
Nope I checked with Philip who is teh author of that line and i is fine this way.
The reason is that I added test which runs prior to this one and which creates a spark context, in which case this test fails with too many spark contexts

Reply:
I think right now we run each individual test file separately https://github.com/databricks/spark-deep-learning/blob/master/python/run-tests.sh#L100-L109 in our Travis CI.
(Correct me if I am wrong) I think creating new ones or not shouldn't matter.

Reply:
It matters when you run the tests locally

Reply:
^ I agree. When testing locally and running multiple tests simultaneously, there will be conflict. What I was trying to say is that I support the way you are changing the spark context.

Reply:
Can you guys explain this to me? How are you all running this locally? Each test suite should create and destroy it's spark context. Is that not currently happening?

Reply:
Ok it might not be dependence between tests. 

named_image.py::_getScaleHintList() calls into jvm to get the list of valid resize hints so it creates a spark context. And it gets called before Philip's setup_env. 

I am not sure what's the proper practice here, should I destroy it after use?

Reply:
In general it's not safe to assume that a SparkContext is available on module import. I don't remember why, but I know it's bit us before. That's why we end up with code like this, https://github.com/graphframes/graphframes/pull/169/files#diff-e81e6b169c0aa35012a3263b2f31b330R385

In this case we might be able to avoid using `_getScaleHintList()` if we inherit from `JavaTransformer`.

Reply:
I don't think inheriting from JavaTransformer changes things here. I still need a Param with validator on the python side.

I guess technically we don't need parameter validator on the python side and just use the one on the scala side. Except then you get the exception only after running transform and not when you set the parameter which is not nice. 

Or I could put initialization into constructor.

Reply:
Wait ... does the validator on scala Param not get run until stransform is called? I might look into fixing that after this release.

Reply:
The params are not transferred to scala until you call _transform. At least that how it looks to me after a quick look. I can transfer them eagerly in setParams.

Reply:
O both work. You can either 
1) leave validation to scala and eagerly transfer params  
or 
2) add converter with lazy initialization.

Reply:
I think within mllib `lazy` initialization is mostly used, I'm not sure why that's the case.